### DISCUSSION TITLE
why was npx added in this commit?

### DISCUSSION THREAD

**Discussion by Zamiell** [typescript-eslint/typescript-eslint#8491] [Technical Discussions]
https://github.com/typescript-eslint/typescript-eslint/commit/dc6f50c2a7cde21c15655442cbf6ebaaff3d06d7

npx shouldn't be necessary, thus I do not understand this commit, and it doesn't seem to be documented anywhere

**Comment by bradzacher** [comment]
Yarn2+ is a bit funny... It doesn't let you use binaries declared in the workspace root. So the correct fix here would have been declaring the version instead so that yarn could have worked.

**Reply by Zamiell** [reply]
oh, i see, thanks.
it sounds like then just `npm run foo` would still work regardless.
(i guess as long as yarn is configured to use legacy node_modules instead of pnp, which [this repo is](https://github.com/typescript-eslint/typescript-eslint/blob/main/.yarnrc.yml#L2).)
is there any advantage to using yarn to launch scripts rather than npm?
just consistency i guess since the project is "supposed" to be using yarn?

**Reply by bradzacher** [reply]
Yeah that's it - we don't hugely want to mix n match things if we can avoid it.
But I do realise the irony in saying that as this commit mixes and matches things haha

We should probably just fix the deps up properly so that we have all the "binaries" declared in each package so that we don't have this issue again.
